OloEngine/src/OloEngine/Scripting/Lua/LuaScriptEngine.cpp

```
@@ -83,34 +79,131 @@ namespace OloEngine
+            OLO_CORE_ERROR("[LuaScriptEngine] Error executing '{}': {}", scriptFile, error.what());
+            return;
+        }
+
+        // Expect the script to return a table
+        if (result.get_type() != sol::type::table)
+        {
+            OLO_CORE_WARN("[LuaScriptEngine] Script '{}' did not return a table — lifecycle callbacks will not work", scriptFile);
+            return;
+        }
+
+        sol::table scriptTable = result;
```

Clang (14.0.6)

[warning] variable 'scriptTable' is not initialized

(cppcoreguidelines-init-variables)